src/System.Private.CoreLib/shared/System/Type.cs

```diff
@@ -26,10 +27,10 @@ public abstract partial class Type : MemberInfo, IReflect
         public abstract new Module Module { get; }
 
         public bool IsNested => DeclaringType != null;
-        public override Type DeclaringType => null;
-        public virtual MethodBase DeclaringMethod => null;
+        public override Type? DeclaringType => null;
```

Comment:
We need to review all members for nullability, not just those the tool flags.  In particular, the tool currently isn't analyzing `abstract` members as far as I can tell, and limited on `virtuals`, because it doesn't have all of the implementations to analyze.  @safern, can you update the tool to at least find all implementations in the same assembly and factor in overrides as part of the analysis?

In particular, for example, `Namespace` above can absolutely return `null`, e.g. for a type that's not in a namespace and just at the global scope.  That becomes obvious when looking at derivations, e.g.
https://github.com/dotnet/coreclr/blob/378a0424d350911b569ff3ab0263bb1c1e5fa5ba/src/System.Private.CoreLib/src/System/RtType.cs#L3316-L3328

When annotating a base type like this, I suggest we also annotate all derived types in the same assembly as part of the same PR, e.g. RuntimeType should be included in this PR.  Otherwise, we're not getting a full picture of the type's surface area.  We also need to be thorough about reviewing all of the surface area, even if the tool doesn't touch it.

Reply:
Yeah, makes sense, it makes it really hard to review the annotations were added correctly without derived types. I opened: https://github.com/safern/ApplyNullableDecorators/issues/11 in my tool. Will try and get it fixed ASAP.